tests/CommunityToolkit.Aspire.Hosting.Rust.Tests/AppHostTests.cs

```diff
@@ -0,0 +1,21 @@
+using CommunityToolkit.Aspire.Testing;
+using FluentAssertions;
+
+namespace CommunityToolkit.Aspire.Hosting.Rust.Tests;
+
+#pragma warning disable CTASPIRE001
+public class AppHostTests(AspireIntegrationTestFixture<Projects.CommunityToolkit_Aspire_Hosting_Rust_AppHost> fixture) : IClassFixture<AspireIntegrationTestFixture<Projects.CommunityToolkit_Aspire_Hosting_Rust_AppHost>>
+{
+    [Fact]
+    public async Task ResourceStartsAndRespondsOk()
+    {
+        var appName = "rust-app";
+        var httpClient = fixture.CreateHttpClient(appName);
+
+        await fixture.App.WaitForTextAsync("listening on", appName).WaitAsync(TimeSpan.FromMinutes(5));
```

Comment:
You could make your app host resource have a http health check and then use the resource notification service to wait for it to start.

We're better off moving towards that rather than using this method.

Reply:
We use `WaitForTextAsync` in `dotnet/aspire` a lot.
https://github.com/dotnet/aspire/blob/cc15725475b79e0ebd9e6bc9e0a0570429460817/tests/Aspire.Hosting.Redis.Tests/RedisFunctionalTests.cs#L79

https://github.com/dotnet/aspire/blob/cc15725475b79e0ebd9e6bc9e0a0570429460817/tests/Aspire.Hosting.PostgreSQL.Tests/PostgresFunctionalTests.cs#L167

https://github.com/dotnet/aspire/blob/cc15725475b79e0ebd9e6bc9e0a0570429460817/tests/Aspire.Hosting.Nats.Tests/NatsFunctionalTests.cs#L37